CutBy Element

OzanOzanOzan opened the pull request:
- CutBy Element Implementation 
    - Holes 
    - Slot 
    - HoleCatalogue 
    
- Aveva midship was  2.8.5 version, it is replaced with new model of 2.8.6

ToDo for future:
   - HoleShapes from parametric types ( Rectangular etc.)

OzanOzanOzan committed:
Read CutBy element of OCX file

OzanOzanOzan committed:
Adjust the new files to new folder structure

paulbuechner commented:
Hey @OzanOzanOzan, we appreciate your effort in contributing to this project!

I have some adjustments to your PR in the pipeline, to make your changes more in line with of how we do things, like building and adding to the assembly, as the current state is already a bit messy and we don't want to complicate things even more.

I will push them on top of this PR after clarifying the changes with @skfcz 

Changes I made:
- Add `CutBy` to the assembly
- code cleanup

![foo](https://user-images.githubusercontent.com/45827409/231520424-7cbfe1db-8025-48d3-a69d-7569f9fa7d1a.png)

paulbuechner committed:
feat: append `CutBy` to assembly

* chore: fix header guards

* refactor: `ocx-cut-by` into shared reader
- `cutBy` is referenced from multiple entities such as `Panel` and `Plate` therefore it makes sense to make it a shared reader

* chore: minor improvments

paulbuechner committed:
chore: update README

- add `OzanOzanOzan` to list of contributors

* chore: unofficially remove support for SHIPXML

OzanOzanOzan commented:
Hi @paulbuechner ,

I want to thank you as well for initiating this project which is indeed exciting.
I can see that things are changing quickly in this front, when I have cloned the branch it was a bit different architecture.
I dont have anything against with your changes, you are the architect of this projcet after all. :)
I will try to contribute the project as much as I can in future.
Thanks again.

Ozan

paulbuechner commented:
@OzanOzanOzan Architecture wise should be the same... I don't know when you forked it, but I did make some changes to the folder structure, to make it easier (IMO) to get into the workings of the reading mechanism. I adopted your added feature to the structure so it stays consistent.

Anyway, nice to hear you want to continue contributing to this project. We really appreciate it. Also, don't hesitate to start a discussion if you have any suggestions on how the `ocx` lib architecture / API can be improved. This is definitely not final (hence no official release) and major changes should be no problem iterating over.
